game-core/src/test/java/games/strategy/engine/data/properties/MapPropertyTest.java

```diff
@@ -23,27 +23,32 @@
 
     @Test
     void shouldReturnTrueWhenValueIsMapAndPropertyMapIsEmpty() {
-      final MapProperty<Integer> mapProperty = newMapProperty(ImmutableMap.of());
+      final MapProperty<Integer> mapProperty = newMapProperty(Map.of());
 
-      assertThat(mapProperty.validate(ImmutableMap.of(KEY_1, "11", KEY_2, "22")), is(true));
+      assertThat(mapProperty.validate(Map.of(KEY_1, "11", KEY_2, "22")), is(true));
     }
 
     @Test
     void shouldReturnTrueWhenValueIsMapAndPropertyMapContainsNullKey() {
-      final MapProperty<Integer> mapProperty = newMapProperty(Collections.singletonMap(null, 42));
+      final Map<String, Integer> map = new HashMap<>();
+      map.put(null, 42);
+
+      final MapProperty<Integer> mapProperty = newMapProperty(map);
 
-      assertThat(mapProperty.validate(ImmutableMap.of(KEY_1, 11, KEY_2, 22)), is(true));
+      assertThat(mapProperty.validate(Map.of(KEY_1, 11, KEY_2, 22)), is(true));
     }
 
     @Test
     void shouldReturnTrueWhenValueIsMapAndEmpty() {
-      assertThat(mapProperty.validate(ImmutableMap.of()), is(true));
+      assertThat(mapProperty.validate(Map.of()), is(true));
     }
 
     @Test
     void shouldReturnTrueWhenValueIsMapAndKeysAreCompatibleAndValuesAreCompatible() {
-      assertThat(mapProperty.validate(ImmutableMap.of(KEY_1, 11, KEY_2, 22)), is(true));
-      assertThat(mapProperty.validate(Collections.singletonMap(null, 33)), is(true));
+      assertThat(mapProperty.validate(Map.of(KEY_1, 11, KEY_2, 22)), is(true));
+      final Map<String, Integer> map = new HashMap<>();
+      map.put(null, 33);
+      assertThat(mapProperty.validate(map), is(true));
```

Comment:
Alternative approach:
The script does allow the old methods if null is hardcoded

Reply:
Maybe, the null hardcoded would lead to false positives. Please note the script is not yet enabled. The test for PointFileReaderWriter has major problems, I'll need your help likely to understand why the test is expected to pass to begin with and seems to depend on non-deterministic ordering.

`null` in a collection is a smell, I suspect the actual code smell here is a 'data clump' code smell. Fixing that IMO is the right fix, not hacking the script.

Reply:
It'd be easier to instead to define an annotation that skips custom style checks. If we do so, then I'm concerned the conventions become toothless, in which case what was the point of defining them?

Reply:
IE: conventions == "do this unless you really can't", vs "Please do this if it's convenient and something you already agree with". Conventions are meant to make people do things that they don't necessarily agree with for consistency of the project. Not everyone agrees with 2 space indents or line length. As Ron mentioned, better to be consistently bad vs inconsistently good.